crates/cairo-profiler/src/trace_reader.rs

```diff
@@ -125,28 +125,32 @@ pub fn collect_samples_from_trace(trace: &CallTrace) -> Vec<Sample> {
     samples
 }
 
-pub struct ResourcesKeys {
-    pub builtins: HashSet<String>,
-    pub syscalls: HashSet<DeprecatedSyscallSelector>,
-}
+pub struct SampleUnits(HashSet<String>);
 
-impl ResourcesKeys {
-    pub fn measurement_types(&self, context: &mut ProfilerContext) -> Vec<ValueType> {
-        let mut value_types = vec![];
+impl SampleUnits {
+    pub fn new(mut units: HashSet<String>) -> Self {
+        units.extend([
+            String::from("calls"),
+            String::from("n_steps"),
+            String::from("n_memory_holes"),
+        ]);
+        Self(units)
+    }
 
-        for builtin in &self.builtins {
-            let unit_string = " ".to_string().add(&builtin.replace('_', " "));
-            value_types.push(ValueType {
-                r#type: context.string_id(builtin).into(),
-                unit: context.string_id(&unit_string).into(),
-            });
-        }
-        for syscall in &self.syscalls {
-            let type_string = format!("{syscall:?}");
-            let unit_string = " ".to_string().add(&type_string);
+    pub fn pprof_sample_units(&self, context: &mut ProfilerContext) -> Vec<pprof::ValueType> {
+        let mut value_types = vec![];
 
-            value_types.push(ValueType {
-                r#type: context.string_id(&type_string).into(),
+        for unit in &self.0 {
+            let unit_without_underscores = unit.replace('_', " ");
+            let unit_without_prefix = if unit_without_underscores.starts_with("n ") {
+                unit_without_underscores.strip_prefix("n ").unwrap()
+            } else {
+                &unit_without_underscores
+            };
```

Comment:
I'm not a big fan of this code 😅 Would it be possible to remove it entirely? Maybe by removing `n_` from the `SampleUnits::new` (not sure if possible)